### PR TITLE
RHEL/Fedora now defaults to setting crashkernel

### PR DESCRIPTION
**Pull request by jelly**
This does not save us a lot of time, as even on RHEL we need to re-generate the initrd however we do skip one reboot which can be flaky.

But what is a nice side effect is that we now test the Distro default crashkernel setting.

### CODE REVIEW

File: test/verify/check-kdump [hunk 1]
```diff
@@ -68,7 +70,8 @@ class TestKdump(KdumpHelpers):
         m = self.machine
 
         b.wait_timeout(120)
-        m.execute("systemctl enable kdump; systemctl start kdump || true")
+        if m.image.startswith("fedora"):
+            m.execute("systemctl enable kdump; systemctl start kdump || true")
```
Comment (martinpitt) [inline test/verify/check-kdump:73]: Can we lose the `|| true` here? I hate these, they just make tests unpredictable later on. Also, `enable --now`

File: test/verify/check-kdump [hunk 2]
```diff
@@ -68,7 +70,8 @@ class TestKdump(KdumpHelpers):
         m = self.machine
 
         b.wait_timeout(120)
-        m.execute("systemctl enable kdump; systemctl start kdump || true")
+        if m.image.startswith("fedora"):
```
Comment (martinpitt) [inline test/verify/check-kdump:72]: let's add an `else` which asserts that the service is already running (on RHEL), to save us from surprises?

File: test/verify/check-kdump [hunk 3]
```diff
@@ -24,9 +24,11 @@ from lib.constants import TEST_OS_DEFAULT
 class KdumpHelpers(testlib.MachineCase):
     def enableKdump(self):
```
Comment (martinpitt) [inline test/verify/check-kdump:25]: This helper contains so much unrelated junk.. `start_cockpit()` and the lines below are completely unnecessary (done through `self.login_and_go()` already), the `allow_restart()` should move into setUp() or the tests as that is unrelated. It's ok for now to always call the helper, although IMHO the tests would be easier to understand if they moved out the condition -- "if f38: enable_kdump()". but that would duplicate it between the two test classes, so keep it for now.

File: test/verify/check-kdump [hunk 4]
```diff
@@ -148,7 +151,7 @@ class TestKdump(KdumpHelpers):
         self.crashKernel("copied through SSH to root@localhost:/var/crash as vmcore", cancel=True)
 
         # we should have the amount of memory reserved that we indicated
```
Comment (martinpitt) [inline test/verify/check-kdump:152]: s/we indicated/crashkernel=auto defaults to for our VM RAM size/

File: test/verify/check-kdump [hunk 5]
```diff
@@ -417,7 +420,8 @@ class TestKdumpNFS(KdumpHelpers):
         self.machines["nfs"].execute("mkdir -p /srv/kdump/var/crash; firewall-cmd --add-service nfs; systemctl restart nfs-server")
 
         # set up client machine
-        m.execute("systemctl disable kdump")
+        if m.image.startswith("fedora"):
+            m.execute("systemctl disable kdump")
```
Comment (martinpitt) [inline test/verify/check-kdump]: This feels a bit arbitrary. I thought Fedora already doesn't enable kdump.service by default?

File: test/verify/check-kdump [hunk 6]
```diff
@@ -419,11 +418,10 @@ class TestKdumpNFS(KdumpHelpers):
         self.machines["nfs"].write("/etc/exports", "/srv/kdump 10.111.113.0/24(rw,no_root_squash)\n")
         self.machines["nfs"].execute("mkdir -p /srv/kdump/var/crash; firewall-cmd --add-service nfs; systemctl restart nfs-server")
 
+        # there shouldn't be any crash reports in the target directory
+        self.assertEqual(m.execute("find /var/crash -maxdepth 1 -mindepth 1 -type d"), "")
+
         # set up client machine
-        if m.image.startswith("fedora"):
-            m.execute("systemctl disable kdump")
```
Comment (martinpitt) [inline test/verify/check-kdump]: Ah, that addresses my earlier comment. Please reverse the commit order or squash.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! I'm glad to see these cleanups.

File: test/verify/check-kdump (hunk 5 above)
Comment (jelly) [reply test/verify/check-kdump]: Dropped it in the follow up commit as it is indeed useless.

File: test/verify/check-kdump (hunk 3 above)
Comment (jelly) [reply test/verify/check-kdump:25]: Right, agreed. Let's do that in a future cleanup when F38 is dropped.

Comment (martinpitt) [reply test/verify/check-kdump:25]: I meant "keep the condition"; Please drop the other junk.

File: test/verify/check-kdump [hunk 7]
```diff
@@ -68,7 +70,10 @@ class TestKdump(KdumpHelpers):
         m = self.machine
 
         b.wait_timeout(120)
-        m.execute("systemctl enable kdump; systemctl start kdump || true")
+        if m.image.startswith("fedora"):
+            m.execute("systemctl enable --now kdump")
+        else:
+            m.execute("systemctl is-enabled kdump")
```
Comment (martinpitt) [inline test/verify/check-kdump:75]: `is-active` is more interesting

File: test/verify/check-kdump [hunk 8]
```diff
@@ -421,8 +420,10 @@ class TestKdumpNFS(KdumpHelpers):
         self.machines["nfs"].write("/etc/exports", "/srv/kdump 10.111.113.0/24(rw,no_root_squash)\n")
         self.machines["nfs"].execute("mkdir -p /srv/kdump/var/crash; firewall-cmd --add-service nfs; systemctl restart nfs-server")
 
-        # ensure there is no local /var/crash, should not break kdump
-        m.execute("rmdir /var/crash")
+        # there shouldn't be any crash reports in the target directory
```
Comment (martinpitt) [inline test/verify/check-kdump:423]: This "clean up NFS test case" commit message is now obsolete: It doesn't "drop the disabling of kdump.service" any more.

**Review by martinpitt — CHANGES_REQUESTED** [review]

**Review by martinpitt — APPROVED** [review]
Great, thanks! (Tests are still running, assuming green)

**Comment by martinpitt** [comment]
Interestting, `is-active kdump.service` fails on [c8s](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19473-20231013-100021-74eecadc-centos-8-stream-expensive/log.html#59-1) and [r93](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19473-20231013-100021-74eecadc-rhel-9-3-expensive/log.html#65-1). This is apparently flaky, as it worked in the first attempt. Perhaps it's still "activating" or something like that?

**Comment by jelly** [comment]
> Interestting, `is-active kdump.service` fails on [c8s](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19473-20231013-100021-74eecadc-centos-8-stream-expensive/log.html#59-1) and [r93](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19473-20231013-100021-74eecadc-rhel-9-3-expensive/log.html#65-1). This is apparently flaky, as it worked in the first attempt. Perhaps it's still "activating" or something like that?

Likely, let's make it non-fatal and wait.

**Comment by martinpitt** [comment]
Not sure what to make of [this failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19473-20231013-115753-ba16f963-rhel-9-4-expensive/log.html#65), but it could be a "normal" timeout flake, so worth retrying.

**Review by martinpitt — APPROVED** [review]
LGTM if/once green.

**Comment by jelly** [comment]
> Not sure what to make of [this failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19473-20231013-115753-ba16f963-rhel-9-4-expensive/log.html#65), but it could be a "normal" timeout flake, so worth retrying.

The kdump tests are very sensitive to test infra being under load.

**Pull request merged**